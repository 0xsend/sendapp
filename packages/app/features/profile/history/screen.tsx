import { Fade, Paragraph, Spinner, Stack, Text, XStack, YStack } from '@my/ui'
import { useUser } from 'app/utils/useUser'
import { AvatarProfile, type AvatarProfileProps } from '../AvatarProfile'
import { useInterUserActivityFeed } from '../utils/useInterUserActivityFeed'
import type { Activity } from 'app/utils/zod/activity'
import { amountFromActivity } from 'app/utils/activity'
import { FlatList } from 'react-native'
import { ActivityDetails } from 'app/features/activity/ActivityDetails'
import SendButton from '../ProfileButtons'
import ProfileHeader from '../components/ProfileHeader'
import { useProfileLookup } from 'app/utils/useProfileLookup'
import { IconArrowRight } from 'app/components/icons'
import {
  isTemporalEthTransfersEvent,
  isTemporalTokenTransfersEvent,
} from 'app/utils/zod/activity/TemporalTransfersEventSchema'
import { useProfileScreenParams, useRootScreenParams } from 'app/routers/params'
import { ShareOtherProfileDialog } from '../components/ShareOtherProfileDialog'
import { useActivityDetails } from 'app/provider/activity-details'
import type { PropsWithChildren } from 'react'

interface ProfileScreenProps {
  sendid?: number | null
}

export function ProfileHistoryScreen({ sendid: propSendid }: ProfileScreenProps) {
  const [{ sendid: paramSendid }] = useProfileScreenParams()
  const otherUserId = propSendid || Number(paramSendid)
  const {
    data: otherUserProfile,
    isLoading,
    error,
  } = useProfileLookup('sendid', otherUserId?.toString() || '')
  const { user, profile: currentUserProfile } = useUser()
  const [{ profile: profileParam }, setRootScreenParams] = useRootScreenParams()
  const { selectActivity, isOpen } = useActivityDetails()

  const {
    data,
    isLoading: isLoadingActivities,
    error: activitiesError,
    isFetchingNextPage: isFetchingNextPageActivities,
    fetchNextPage,
    hasNextPage,
  } = useInterUserActivityFeed({
    pageSize: 10,
    otherUserId,
    currentUserId: currentUserProfile?.send_id,
  })
  const { pages } = data ?? {}
  const activities = pages?.flat() || []

  if (isLoading) {
    return (
      <Stack w="100%" h="100%" jc={'center'} ai={'center'} f={1} gap="$6">
        <Spinner size="large" color="$primary" />
      </Stack>
    )
  }
  if (error) {
    return (
      <Stack w="100%" h="100%" jc={'center'} ai={'center'} f={1} gap="$6">
        <Text theme="red" color={'$color8'}>
          {error.message}
        </Text>
      </Stack>
    )
  }
  return (
    <XStack w={'100%'} gap={'$4'} height={'100%'}>
      <YStack
        f={1}
        height={'100%'}
        gap={'$2'}
        display={profileParam || isOpen ? 'none' : 'flex'}
        $gtLg={{
          display: 'flex',
          maxWidth: '50%',
          pb: '$3.5',
        }}
      >
        {activitiesError !== null && (
          <YStack f={1}>
            <Paragraph
              maxWidth={600}
              fontFamily={'$mono'}
              fontSize={'$5'}
              theme={'red'}
              color={'$color8'}
              mt={'$4'}
              ta={'center'}
            >
              {activitiesError?.message.split('.').at(0) ?? `${activitiesError}`}
            </Paragraph>
          </YStack>
        )}
        {Boolean(otherUserProfile) && (
          <>
            <ProfileHeader profile={otherUserProfile} />
            <Stack f={1} mt={'$2'}>
              {activities?.length ? (
                <FlatList
                  style={{ flex: 1 }}
                  data={activities}
                  keyExtractor={(activity) =>
                    `${activity.event_name}-${activity.created_at}-${activity?.from_user?.id}-${activity?.to_user?.id}`
                  }
                  renderItem={({ item: activity, index }) => {
                    const date = activity.created_at.toLocaleDateString()
                    const nextDate = activities[index + 1]?.created_at.toLocaleDateString()
                    const shouldShowDatePill = !nextDate || date !== nextDate

                    return (
                      <>
                        <Fade>
                          <TransactionEntry
                            activity={activity}
                            sent={activity?.to_user?.id !== user?.id}
                            otherUserProfile={otherUserProfile}
                            currentUserProfile={currentUserProfile}
                            onPress={() => selectActivity(activity)}
                          />
                        </Fade>
                        {shouldShowDatePill ? <DatePill date={date} /> : null}
                      </>
                    )
                  }}
                  onEndReached={() => fetchNextPage()}
                  ListEmptyComponent={
                    !isLoadingActivities && isFetchingNextPageActivities ? (
                      <Spinner size="small" color={'$color12'} my={'$4'} />
                    ) : null
                  }
                  ListFooterComponent={
                    hasNextPage ? <Spinner size="small" color={'$color12'} my={'$4'} /> : null
                  }
                  inverted={true}
                  showsVerticalScrollIndicator={false}
                />
              ) : (
                <YStack f={1}>
                  <Paragraph
                    size={'$8'}
                    fontWeight={'300'}
                    color={'$color10'}
                    textAlign={'center'}
                    pt={'$1.5'}
                  >
                    No Activities
                  </Paragraph>
                </YStack>
              )}
            </Stack>
            <SendButton
              identifier={otherUserProfile?.tag ?? otherUserProfile?.sendid ?? ''}
              idType={otherUserProfile?.tag ? 'tag' : 'sendid'}
            />
          </>
        )}
      </YStack>
      {isOpen ? (
        <ActivityDetails
          w={'100%'}
          $gtLg={{
            maxWidth: '47%',
          }}
        />
      ) : (
        <ShareOtherProfileDialog
          profile={otherUserProfile}
          isOpen={profileParam !== undefined}
          onClose={() => setRootScreenParams({ profile: undefined }, { webBehavior: 'replace' })}
        />
      )}
    </XStack>
  )
}

const TransactionEntry = ({
  activity,
  sent,
  otherUserProfile,
  currentUserProfile,
  onPress,
}: {
  activity: Activity
  sent: boolean
  otherUserProfile?: AvatarProfileProps
  currentUserProfile?: AvatarProfileProps
  onPress: () => void
}) => {
  const {
    data: { note },
  } = activity
  const amount = amountFromActivity(activity)
  const date = useTransactionEntryDate({ activity, sent })

  return (
    <XStack justifyContent={sent ? 'flex-end' : 'flex-start'} testID="activityTest" my={'$2.5'}>
      <YStack gap={'$1'}>
        <YStack
          bg={'$color1'}
          p={'$4'}
          br={'$4'}
          maxWidth={300}
          gap={'$3'}
          ai={sent ? 'flex-end' : 'flex-start'}
          onPress={onPress}
          cursor={'pointer'}
        >
          <XStack
            gap={'$3'}
            ai={'center'}
            fd={sent ? 'row-reverse' : 'row'}
            style={{ width: 'max-content' }}
            alignSelf={sent ? 'flex-end' : 'flex-start'}
          >
            <AvatarProfile
              profile={sent ? currentUserProfile : otherUserProfile}
              mx={0}
              size="$5"
            />
            <YStack>
              <XStack gap={'$2'} ai={'center'} fd={sent ? 'row-reverse' : 'row'}>
                {!sent && <IconArrowRight size={'$0.9'} rotate={'90deg'} color={'$olive'} />}
                <Paragraph size={'$3'} color={'$color8'} theme={sent ? 'red' : 'green'}>
                  You {sent ? 'Sent' : 'Received'}
                </Paragraph>
                {sent && <IconArrowRight size={'$0.9'} rotate={'-90deg'} color={'$red10Dark'} />}
              </XStack>
              <Paragraph size={'$7'}>{amount}</Paragraph>
            </YStack>
          </XStack>
          {note && (
            <Paragraph
              fontSize={17}
              color={'$silverChalice'}
              w={'100%'}
              whiteSpace={'pre-wrap'}
              $theme-light={{
                color: '$darkGrayTextField',
              }}
            >
              {decodeURIComponent(note)}
            </Paragraph>
          )}
        </YStack>
        <XStack jc={sent ? 'flex-end' : 'flex-start'}>{date}</XStack>
      </YStack>
    </XStack>
  )
}

const DatePill = ({ date }: { date: string }) => {
  return (
    <Paragraph
      ff={'$mono'}
      textAlign={'center'}
      size={'$4'}
      py={'$0.25'}
      bc={'$background'}
      color={'$color10'}
      px={'$0.9'}
      br={'$2'}
    >
      {date}
    </Paragraph>
  )
}

const useTransactionEntryDate = ({ activity, sent }: { activity: Activity; sent: boolean }) => {
  const { created_at, data } = activity
  const isTemporalTransfer =
    isTemporalEthTransfersEvent(activity) || isTemporalTokenTransfersEvent(activity)

  if (isTemporalTransfer) {
    switch (data.status) {
      case 'failed':
      case 'cancelled':
        return <DateText sent={sent}>Failed</DateText>
      case 'confirmed':
        return (
          <DateText sent={sent}>
            {new Date(created_at).toLocaleTimeString([], { hour: 'numeric', minute: '2-digit' })}
          </DateText>
        )
      default:
        return <Spinner size="small" color={'$color11'} />
    }
  }

  return (
    <DateText sent={sent}>
      {new Date(created_at).toLocaleTimeString([], { hour: 'numeric', minute: '2-digit' })}
    </DateText>
  )
}

const DateText = ({ children, sent }: PropsWithChildren & { sent: boolean }) => {
  return (
    <Paragraph
      display={'flex'}
      size={'$2'}
      ta={sent ? 'right' : 'left'}
      color={'$color4'}
      $theme-light={{ color: '$silverChalice' }}
    >
      {children}
    </Paragraph>
  )
}
