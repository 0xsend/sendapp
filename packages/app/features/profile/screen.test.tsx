import { test } from '@jest/globals'
import { ProfileScreen } from './screen'
import { TamaguiProvider, config } from '@my/ui'
import { render, screen, act, userEvent } from '@testing-library/react-native'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

const queryClient = new QueryClient()
const TAG_NAME = 'pip_test44677'
const PROFILE = {
  avatar_url:
    'https://fjswgwdweohwejbrmiil.supabase.co/storage/v1/object/public/avatars/db59dfd6-16e6-4c30-8337-4bb33905828f/1697315518383.jpeg',
  name: 'Mabel Bechtelar',
  about:
    'Doctissimae poster est quibus solut quae concuriosum quod, disputatur sit voluptate ea interror pugnantium est conspecta.',
  referral_code: 'p2us75d2560',
  tag_name: TAG_NAME,
  address: '0x3D0B692e4b10A6975658808a6DB9F56C89d3d4a4',
  chain_id: 845337,
}

jest.mock('solito', () => ({
  useRoute: () => ({ params: { tag: TAG_NAME } }),
  createParam: jest.fn().mockReturnValue({
    useParam: jest.fn().mockReturnValue([TAG_NAME]),
  }),
}))

jest.mock('app/utils/useProfileLookup', () => ({
  useProfileLookup: jest.fn().mockReturnValue({
    data: PROFILE,
    error: null,
  }),
}))

jest.mock('app/utils/useUser', () => ({
  useUser: jest.fn().mockReturnValue({
    user: {
      id: '1',
    },
  }),
}))

jest.mock('app/utils/send-accounts')

jest.mock('wagmi')

jest.mock('@my/wagmi', () => {
  const originalModule = jest.requireActual<typeof import('@my/wagmi')>('@my/wagmi')
  return {
    __esModule: true, // Use it when dealing with esModules
    ...originalModule,
    baseMainnetClient: {
      transport: {
        url: 'http://127.0.0.1',
      },
    },
    baseMainnetBundlerClient: {
      transport: {
        url: 'http://127.0.0.1',
      },
    },
  }
})

test('ProfileScreen', async () => {
  jest.useFakeTimers()

  render(
    <QueryClientProvider client={queryClient}>
      <TamaguiProvider defaultTheme={'dark'} config={config}>
        <ProfileScreen />
      </TamaguiProvider>
    </QueryClientProvider>
  )

  await act(() => jest.runAllTimers())

  const h1 = screen.getByText(PROFILE.name)
  expect(h1).toBeOnTheScreen()
  const h2 = screen.getByText(`@${PROFILE.tag_name}`)
  expect(h2).toBeOnTheScreen()
  const paragraph = screen.getByText(PROFILE.about)
  expect(paragraph).toBeOnTheScreen()
  const avatar = screen.getByTestId('avatar')
  expect(avatar).toBeOnTheScreen()
  const image = screen.getByRole('image', { name: PROFILE.name })
  expect(image).toBeOnTheScreen()
  expect(image.props.source).toStrictEqual({
    uri: PROFILE.avatar_url,
  })
  const button1 = screen.getByTestId('openSendDialogButton')
  expect(button1).toBeOnTheScreen()
  const button2 = screen.getByText('Request')
  expect(button2).toBeOnTheScreen()
  expect(screen.toJSON()).toMatchSnapshot('ProfileScreen')

  await act(async () => {
    // await user.press(button1) // @note this does not work
    button1.props.onPress()
    jest.runAllTimers()
  })

  const dialog = screen.getByTestId('sendDialogContainer')
  expect(dialog).toBeOnTheScreen()
})
