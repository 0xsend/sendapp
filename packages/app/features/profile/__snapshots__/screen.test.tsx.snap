// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen: ProfileScreen 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "gap": 18,
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "gap": 7,
        "height": "100%",
        "overflow": "hidden",
      }
    }
  >
    <View
      cancelable={true}
      disabled={false}
      focusable={true}
      minPressDuration={0}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onPress={[Function]}
      onPressIn={[Function]}
      onPressOut={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#111f22",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
          "position": "relative",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 24,
            "width": "80%",
          }
        }
      >
        <View
          cancelable={[Function]}
          disabled={false}
          focusable={[Function]}
          minPressDuration={0}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 9,
              "borderBottomRightRadius": 9,
              "borderTopLeftRadius": 9,
              "borderTopRightRadius": 9,
              "flexDirection": "column",
              "height": 64,
              "justifyContent": "center",
              "maxHeight": 64,
              "maxWidth": 64,
              "minHeight": 64,
              "minWidth": 64,
              "overflow": "hidden",
              "position": "relative",
              "width": 64,
            }
          }
        >
          <View
            style={
              {
                "bottom": 0,
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 1,
              }
            }
          >
            <Image
              accessibilityLabel="Mabel Bechtelar"
              accessibilityRole="image"
              accessible={true}
              fullscreen={true}
              onError={[Function]}
              onLoad={[Function]}
              source={
                {
                  "uri": "https://fjswgwdweohwejbrmiil.supabase.co/storage/v1/object/public/avatars/db59dfd6-16e6-4c30-8337-4bb33905828f/1697315518383.jpeg",
                }
              }
              style={
                {
                  "height": 64,
                  "width": 64,
                }
              }
              testID="avatarImage"
            />
          </View>
          <View
            style={
              {
                "bottom": 0,
                "flexDirection": "column",
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={64}
              bbWidth={64}
              color="#86AE80"
              focusable={false}
              height={64}
              meetOrSlice={0}
              minX={0}
              minY={0}
              strokeWidth="2"
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "color": "#86AE80",
                  },
                  {
                    "flex": 0,
                    "height": 64,
                    "width": 64,
                  },
                ]
              }
              tintColor="#86AE80"
              vbHeight={24}
              vbWidth={24}
              width={64}
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "strokeWidth",
                  ]
                }
                strokeWidth="2"
              >
                <RNSVGPath
                  d="M12 12C10.9 12 9.95833 11.6083 9.175 10.825C8.39167 10.0417 8 9.1 8 8C8 6.9 8.39167 5.95833 9.175 5.175C9.95833 4.39167 10.9 4 12 4C13.1 4 14.0417 4.39167 14.825 5.175C15.6083 5.95833 16 6.9 16 8C16 9.1 15.6083 10.0417 14.825 10.825C14.0417 11.6083 13.1 12 12 12ZM4 20V17.2C4 16.6333 4.14583 16.1125 4.4375 15.6375C4.72917 15.1625 5.11667 14.8 5.6 14.55C6.63333 14.0333 7.68333 13.6458 8.75 13.3875C9.81667 13.1292 10.9 13 12 13C13.1 13 14.1833 13.1292 15.25 13.3875C16.3167 13.6458 17.3667 14.0333 18.4 14.55C18.8833 14.8 19.2708 15.1625 19.5625 15.6375C19.8542 16.1125 20 16.6333 20 17.2V20H4ZM6 18H18V17.2C18 17.0167 17.9542 16.85 17.8625 16.7C17.7708 16.55 17.65 16.4333 17.5 16.35C16.6 15.9 15.6917 15.5625 14.775 15.3375C13.8583 15.1125 12.9333 15 12 15C11.0667 15 10.1417 15.1125 9.225 15.3375C8.30833 15.5625 7.4 15.9 6.5 16.35C6.35 16.4333 6.22917 16.55 6.1375 16.7C6.04583 16.85 6 17.0167 6 17.2V18ZM12 10C12.55 10 13.0208 9.80417 13.4125 9.4125C13.8042 9.02083 14 8.55 14 8C14 7.45 13.8042 6.97917 13.4125 6.5875C13.0208 6.19583 12.55 6 12 6C11.45 6 10.9792 6.19583 10.5875 6.5875C10.1958 6.97917 10 7.45 10 8C10 8.55 10.1958 9.02083 10.5875 9.4125C10.9792 9.80417 11.45 10 12 10Z"
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <Text
          nativeID="profileName"
          style={
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 27.599999999999998,
              "fontWeight": "600",
              "lineHeight": 28,
              "userSelect": "auto",
              "width": "80%",
            }
          }
          suppressHighlighting={true}
        >
          Mabel Bechtelar
        </Text>
      </View>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        color="#40FB50"
        fill="none"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={0}
        strokeWidth="2"
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#40FB50",
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        tintColor="#40FB50"
        vbHeight={20}
        vbWidth={20}
        width={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "strokeWidth",
            ]
          }
          strokeWidth="2"
        >
          <RNSVGPath
            d="M14.2917 10.625L9.12508 15.7917L10.0001 16.6667L16.6667 10L10.0001 3.33337L9.12508 4.20837L14.2917 9.37504H3.33341V10.625H14.2917Z"
            fill={
              {
                "type": 2,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "flex": 1,
          "paddingBottom": 18,
        }
      }
    >
      <div
        className="css-view-175oi2r r-WebkitOverflowScrolling-150rngu r-flexDirection-eqz5dr r-flexGrow-16y2uox r-flexShrink-1wbh5a2 r-overflowX-11yh6sk r-overflowY-1rnoaur r-transform-173mzie r-scrollbarWidth-2eszeu"
        dir={null}
        onScroll={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onWheel={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <div
          className="css-view-175oi2r"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-position-gtdqiz r-top-ipm5af r-zIndex-mhe3cw"
            dir={null}
          >
            <div
              className="css-view-175oi2r r-transform-173mzie"
              dir={null}
            >
              <View
                cancelable={true}
                collapsable={false}
                disabled={false}
                focusVisibleStyle={{}}
                focusable={[Function]}
                forwardedRef={[Function]}
                jestAnimatedStyle={
                  {
                    "value": {
                      "backgroundColor": "#081619",
                      "borderBottomColor": "transparent",
                      "borderBottomLeftRadius": 9,
                      "borderBottomRightRadius": 9,
                      "borderBottomWidth": 1,
                      "borderLeftColor": "transparent",
                      "borderLeftWidth": 1,
                      "borderRightColor": "transparent",
                      "borderRightWidth": 1,
                      "borderTopColor": "transparent",
                      "borderTopLeftRadius": 9,
                      "borderTopRightRadius": 9,
                      "borderTopWidth": 1,
                      "height": 48,
                      "opacity": 0,
                      "width": "100%",
                    },
                  }
                }
                minPressDuration={0}
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onPress={[Function]}
                onPressIn={[Function]}
                onPressOut={[Function]}
                role="button"
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#081619",
                    "borderBottomColor": "transparent",
                    "borderBottomLeftRadius": 9,
                    "borderBottomRightRadius": 9,
                    "borderBottomWidth": 1,
                    "borderLeftColor": "transparent",
                    "borderLeftWidth": 1,
                    "borderRightColor": "transparent",
                    "borderRightWidth": 1,
                    "borderStyle": "solid",
                    "borderTopColor": "transparent",
                    "borderTopLeftRadius": 9,
                    "borderTopRightRadius": 9,
                    "borderTopWidth": 1,
                    "display": "flex",
                    "flexDirection": "row",
                    "flexWrap": "nowrap",
                    "height": 48,
                    "justifyContent": "center",
                    "minWidth": 150,
                    "opacity": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "width": "100%",
                  }
                }
                testID="profileSendButton"
              >
                <View
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "gap": 7,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    lineBreakMode="clip"
                    numberOfLines={1}
                    style={
                      {
                        "color": "#081619",
                        "flexGrow": 0,
                        "flexShrink": 1,
                        "fontFamily": "System",
                        "fontSize": 16.8,
                        "fontWeight": "600",
                        "lineHeight": 19,
                        "textTransform": "uppercase",
                        "userSelect": "none",
                      }
                    }
                    suppressHighlighting={true}
                  >
                    SEND
                  </Text>
                </View>
              </View>
            </div>
          </div>
          <div
            className="css-view-175oi2r r-flexDirection-1d5kdc7 r-transform-173mzie"
            dir={null}
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              jestAnimatedStyle={
                {
                  "value": {
                    "opacity": 0,
                    "transform": [
                      {
                        "scale": 0.9,
                      },
                    ],
                  },
                }
              }
              style={
                {
                  "opacity": 0,
                  "transform": [
                    {
                      "scale": 0.9,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "marginBottom": 10,
                    "marginTop": 10,
                  }
                }
                testID="activityTest"
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "gap": 2,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "backgroundColor": "#111f22",
                        "borderBottomLeftRadius": 9,
                        "borderBottomRightRadius": 9,
                        "borderTopLeftRadius": 9,
                        "borderTopRightRadius": 9,
                        "flexDirection": "column",
                        "paddingBottom": 18,
                        "paddingLeft": 18,
                        "paddingRight": 18,
                        "paddingTop": 18,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row-reverse",
                          "gap": 13,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 9,
                            "borderBottomRightRadius": 9,
                            "borderTopLeftRadius": 9,
                            "borderTopRightRadius": 9,
                            "flexDirection": "column",
                            "gap": 7,
                            "height": 52,
                            "justifyContent": "center",
                            "marginLeft": "none",
                            "marginRight": "none",
                            "maxHeight": 52,
                            "maxWidth": 52,
                            "minHeight": 52,
                            "minWidth": 52,
                            "overflow": "hidden",
                            "position": "relative",
                            "width": 52,
                          }
                        }
                        testID="avatar"
                      >
                        <View
                          style={
                            {
                              "bottom": 0,
                              "flexDirection": "column",
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "zIndex": 1,
                            }
                          }
                        >
                          <Image
                            accessibilityLabel="??"
                            accessibilityRole="image"
                            accessible={true}
                            fullscreen={true}
                            onError={[Function]}
                            onLoad={[Function]}
                            source={
                              {
                                "uri": "",
                              }
                            }
                            style={
                              {
                                "height": 52,
                                "width": 52,
                              }
                            }
                            testID="avatarImage"
                          />
                        </View>
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "bottom": 0,
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "zIndex": 0,
                            }
                          }
                        >
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight={64}
                            bbWidth={64}
                            color="#86AE80"
                            focusable={false}
                            height={64}
                            meetOrSlice={0}
                            minX={0}
                            minY={0}
                            strokeWidth="2"
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "color": "#86AE80",
                                },
                                {
                                  "flex": 0,
                                  "height": 64,
                                  "width": 64,
                                },
                              ]
                            }
                            tintColor="#86AE80"
                            vbHeight={24}
                            vbWidth={24}
                            width={64}
                          >
                            <RNSVGGroup
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              propList={
                                [
                                  "strokeWidth",
                                ]
                              }
                              strokeWidth="2"
                            >
                              <RNSVGPath
                                d="M12 12C10.9 12 9.95833 11.6083 9.175 10.825C8.39167 10.0417 8 9.1 8 8C8 6.9 8.39167 5.95833 9.175 5.175C9.95833 4.39167 10.9 4 12 4C13.1 4 14.0417 4.39167 14.825 5.175C15.6083 5.95833 16 6.9 16 8C16 9.1 15.6083 10.0417 14.825 10.825C14.0417 11.6083 13.1 12 12 12ZM4 20V17.2C4 16.6333 4.14583 16.1125 4.4375 15.6375C4.72917 15.1625 5.11667 14.8 5.6 14.55C6.63333 14.0333 7.68333 13.6458 8.75 13.3875C9.81667 13.1292 10.9 13 12 13C13.1 13 14.1833 13.1292 15.25 13.3875C16.3167 13.6458 17.3667 14.0333 18.4 14.55C18.8833 14.8 19.2708 15.1625 19.5625 15.6375C19.8542 16.1125 20 16.6333 20 17.2V20H4ZM6 18H18V17.2C18 17.0167 17.9542 16.85 17.8625 16.7C17.7708 16.55 17.65 16.4333 17.5 16.35C16.6 15.9 15.6917 15.5625 14.775 15.3375C13.8583 15.1125 12.9333 15 12 15C11.0667 15 10.1417 15.1125 9.225 15.3375C8.30833 15.5625 7.4 15.9 6.5 16.35C6.35 16.4333 6.22917 16.55 6.1375 16.7C6.04583 16.85 6 17.0167 6 17.2V18ZM12 10C12.55 10 13.0208 9.80417 13.4125 9.4125C13.8042 9.02083 14 8.55 14 8C14 7.45 13.8042 6.97917 13.4125 6.5875C13.0208 6.19583 12.55 6 12 6C11.45 6 10.9792 6.19583 10.5875 6.5875C10.1958 6.97917 10 7.45 10 8C10 8.55 10.1958 9.02083 10.5875 9.4125C10.9792 9.80417 11.45 10 12 10Z"
                                fill={
                                  {
                                    "type": 2,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                      </View>
                      <View
                        style={
                          {
                            "flexDirection": "column",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "flexDirection": "row-reverse",
                              "gap": 7,
                            }
                          }
                        >
                          <Text
                            style={
                              {
                                "color": "hsl(358, 85.3%, 64.0%)",
                                "fontFamily": "System",
                                "fontSize": 15.6,
                                "fontWeight": "300",
                                "lineHeight": 18,
                                "userSelect": "auto",
                              }
                            }
                            suppressHighlighting={true}
                          >
                            You 
                            Sent
                          </Text>
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight={16}
                            bbWidth={16}
                            color="hsl(358, 85.3%, 64.0%)"
                            fill="none"
                            focusable={false}
                            height={16}
                            meetOrSlice={0}
                            minX={0}
                            minY={0}
                            strokeWidth="2"
                            style={
                              [
                                {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                {
                                  "color": "hsl(358, 85.3%, 64.0%)",
                                  "transform": [
                                    {
                                      "rotate": "-90deg",
                                    },
                                  ],
                                },
                                {
                                  "flex": 0,
                                  "height": 16,
                                  "width": 16,
                                },
                              ]
                            }
                            tintColor="hsl(358, 85.3%, 64.0%)"
                            vbHeight={20}
                            vbWidth={20}
                            width={16}
                          >
                            <RNSVGGroup
                              fill={null}
                              propList={
                                [
                                  "fill",
                                  "strokeWidth",
                                ]
                              }
                              strokeWidth="2"
                            >
                              <RNSVGPath
                                d="M14.2917 10.625L9.12508 15.7917L10.0001 16.6667L16.6667 10L10.0001 3.33337L9.12508 4.20837L14.2917 9.37504H3.33341V10.625H14.2917Z"
                                fill={
                                  {
                                    "type": 2,
                                  }
                                }
                                propList={
                                  [
                                    "fill",
                                  ]
                                }
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                        <Text
                          style={
                            {
                              "color": "#FFFFFF",
                              "fontFamily": "System",
                              "fontSize": 24,
                              "fontWeight": "600",
                              "lineHeight": 25,
                              "userSelect": "auto",
                            }
                          }
                          suppressHighlighting={true}
                        >
                          10,000
                        </Text>
                      </View>
                    </View>
                  </View>
                  <Text
                    style={
                      {
                        "color": "#666666",
                        "fontFamily": "System",
                        "fontSize": 14.399999999999999,
                        "fontWeight": "300",
                        "lineHeight": 17,
                        "textAlign": "right",
                        "userSelect": "auto",
                      }
                    }
                    suppressHighlighting={true}
                  >
                    12:00 AM
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                {
                  "backgroundColor": "#081619",
                  "borderBottomLeftRadius": 5,
                  "borderBottomRightRadius": 5,
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "color": "#B2B2B2",
                  "fontFamily": "System",
                  "fontSize": 16.8,
                  "fontWeight": "400",
                  "lineHeight": 19,
                  "paddingBottom": 2,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 2,
                  "textAlign": "center",
                  "userSelect": "auto",
                }
              }
              suppressHighlighting={true}
            >
              5/19/2024
            </Text>
          </div>
        </div>
      </div>
    </View>
  </View>
</View>
`;
